persistent/cPersistence.c

```diff
@@ -190,47 +192,54 @@ ghostify(cPersistentObject *self)
         *dictptr = NULL;
     }
 
-    /* clear all slots besides _p_* */
-    slotnames = pickle_slotnames(Py_TYPE(self));
-    if (slotnames && slotnames != Py_None)
+    /* clear all slots besides _p_*
+     * ( for backward-compatibility reason we do this only if class does not
+     *   override __new__ ) */
+    __new__ = PyObject_GetAttr((PyObject *)Py_TYPE(self), py___new__);
```

Comment:
You don't need and shouldn't use (because overhead) getattr here.  You can just compare the tp_new struct members:

    #ifdef PY3K
        if (self->ob_base.ob_type->tp_new == Pertype.tp_new)
    #else
        if (self->ob_type->tp_new == Pertype.tp_new)
    #endif

This will simplify this change quite a bit, as you won't need the __new__string or to cache the Python version of persistent.__new__.